src/Ushahidi/Modules/V5/Policies/PermissionsPolicy.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Ushahidi\Modules\V5\Policies;
+
+use Ushahidi\Modules\V5\Models\Permissions;
+use App\Auth\GenericUser as User;
+use Ushahidi\Core\Concerns\AdminAccess;
+use Ushahidi\Core\Concerns\PrivAccess;
+use Ushahidi\Core\Concerns\UserContext;
+
+class PermissionsPolicy
+{
+
+
+    use UserContext;
+
+    // It uses `PrivAccess` to provide the `getAllowedPrivs` method.
+    use PrivAccess;
+
+    // Check if user has Admin access
+    use AdminAccess;
+
+    protected $user;
+
+    /**
+     * @param User $user
+     * @return bool
+     */
+    public function index(User $user):bool
+    {
+        $empty_permissions = new Permissions();
+        return $this->isAllowed($empty_permissions, 'search', $user);
+    }
+
+    /**
+     * @param User $user
+     * @param Permissions $permissions
+     * @return bool
+     */
+    public function show(User $user, Permissions $permissions):bool
+    {
+        return $this->isAllowed($permissions, 'read', $user);
+    }
+
+    /**
+     * @param User $user
+     * @param Permissions $permissions
+     * @return bool
+     */
+    public function delete(User $user, Permissions $permissions):bool
+    {
+        return $this->isAllowed($permissions, 'delete', $user);
+    }
+    /**
+     * @param User $user
+     * @param Permissions $permissions
+     * @return bool
+     */
+    public function update(User $user, Permissions $permissions):bool
+    {
+        return $this->isAllowed($permissions, 'update', $user);
+    }
+
+
+    /**
+     * @param User $user
+     * @param Permissions $permissions
+     * @return bool
+     */
+    public function store(User $user):bool
+    {
+        $permissions = new Permissions();
+        return $this->isAllowed($permissions, 'create', $user);
+    }
+
+    /**
+     * @param Permissions $permissions
+     * @param string $privilege
+     * @param user $user
+     * @return bool
+     */
+    public function isAllowed($permissions, $privilege, $user = null):bool
+    {
+        
+        $authorizer = service('authorizer.permission');
+        $user = $authorizer->getUser();
```

Comment:
You don't have to overwrite $user that you passed into the method.

Reply:
this will be changed after merge , we will use the entity